modules/misc/qt.nix

```diff
@@ -194,19 +191,14 @@ in {
 
     home = {
       sessionVariables = envVars;
-      # home.sessionVariables does not support setting the same environment
-      # variable to different values.
-      # Since some other modules may set the QT_PLUGIN_PATH or QML2_IMPORT_PATH
-      # to their own value, e.g.: fcitx5, we avoid conflicts by setting
-      # the values in home.sessionVariablesExtra instead.
-      sessionVariablesExtra = ''
-        export QT_PLUGIN_PATH=${envVarsExtra.QT_PLUGIN_PATH}
-        export QML2_IMPORT_PATH=${envVarsExtra.QML2_IMPORT_PATH}
-      '';
+      sessionSearchVariables = envVarsExtra;
     };
 
     # Apply theming also to apps started by systemd.
-    systemd.user.sessionVariables = envVars // envVarsExtra;
+    systemd.user.sessionVariables = envVars // {
+      QT_PLUGIN_PATH = lib.concatStringsSep ":" envVarsExtra.QT_PLUGIN_PATH;
+      QML2_IMPORT_PATH = lib.concatStringsSep ":" envVarsExtra.QML2_IMPORT_PATH;
+    };
```

Comment:
This suggests the need for `systemd.user.sessionSearchVariables` to me.

Reply:
For a future PR maybe, but not for this one since this already changes too many things.